compiler/types.nim

```diff
@@ -1632,6 +1635,23 @@ proc isTupleRecursive*(t: PType): bool =
   var cycleDetector = initIntSet()
   isTupleRecursive(t, cycleDetector)
 
+proc isRecursivePointer(typ: PType, isPointer: bool): bool =
+  if typ == nil:
+    return false
+  case typ.kind:
+    of tyObject, tyTuple:
+      return isPointer and canFormAcycle(typ)
+    of tyRef, tyPtr:
+      return isRecursivePointer(typ.lastSon, isPointer = true)
+    of tyAlias, tyGenericInst, tyVar, tyLent, tySink, tyArray, tyUncheckedArray, tySequence, tyDistinct:
+      return isRecursivePointer(typ.lastSon, isPointer)
+    else:
+      return false
+
+proc isRecursivePointer*(t: PType): bool =
```

Comment:
you can remove this overload and just defined:
`proc isRecursivePointer(typ: PType, isPointer = false): bool =`

Reply:
Thanks, but I prefer it the way it is.

Reply:
@timotheecour The explanation for why I prefer it this way is, I don't want the `isPointer` parameter on the exported proc `isRecursivePointer`. After all it is just an implementation detail. Somthing that just confused @Araq.